project_task_stage_change_restriction/tests/test_stage_change_restriction.py

```
@@ -0,0 +1,277 @@
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import SUPERUSER_ID
+from odoo.exceptions import UserError
+from odoo.tests import common as tests_common
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+@tests_common.tagged("-at_install", "post_install")
+class TestStageChangeRestriction(BaseCommon):
+    """Validate stage‑change & creation access rules for project tasks."""
```

Ruff (0.11.9)

Docstring contains ambiguous `‑` (NON-BREAKING HYPHEN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)